### PR TITLE
Better serverView exec name; remove SingleServerInventoryView

### PR DESCRIPTION
**Pull request by jihoonson**
### Description

Druid currently has 2 serverViews, regular serverView and filtered serverView. The regular serverView is used to monitor all segment announcements from all data nodes (historicals, tasks, indexers). The filtered serverView is used when you want to watch segment announcements from particular tiers. Since these server views keep track of different sets of druidServers and segments in memory, they should be maintained separately. However, they currently share the same name for their executorService, which can cause confusion and make debugging harder especially in the broker since it is using both serverViews, the filtered view for normal query processing and the regular view to serve the `servers` table (I'm unsure whether this is intended or whether this is a good behavior). This PR changes it to a more obvious name. 

This PR also removes `SingleServerInventoryView`. This view was deprecated a long time ago and has not been documented at least since 0.13 (https://github.com/apache/druid/pull/6127). I also don't think this can be better in any case than `BatchServerInventoryView`. Finally, I merged `AbstractCuratorServerInventoryView` and `BatchServerInventoryView` as we no longer need `AbstractCuratorServerInventoryView` after `SingleServerInventoryView` is removed.

<hr>

##### Key changed/added classes in this PR
 * `BatchServerInventoryViewProvider`/`FilteredBatchServerInventoryViewProvider`
 * `HttpServerInventoryViewProvider`/`FilteredHttpServerInventoryViewProvider`
 * `BatchServerInventoryView`

<hr>

This PR has:
- [x] been self-reviewed.
- [x] been tested in a test Druid cluster.

### CODE REVIEW

File: server/src/main/java/org/apache/druid/client/FilteringSegmentCallback.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+package org.apache.druid.client;
+
+import com.google.common.base.Predicate;
+import org.apache.druid.client.ServerView.CallbackAction;
+import org.apache.druid.client.ServerView.SegmentCallback;
+import org.apache.druid.java.util.common.Pair;
+import org.apache.druid.server.coordination.DruidServerMetadata;
+import org.apache.druid.timeline.DataSegment;
+
+public class FilteringSegmentCallback implements SegmentCallback
```
Comment (abhishekagarwal87) [inline server/src/main/java/org/apache/druid/client/FilteringSegmentCallback.java:29]: I know it is old code but will be nice to have some javadocs here

File: server/src/main/java/org/apache/druid/client/HttpServerInventoryView.java [hunk 2]
```diff
@@ -136,12 +135,12 @@ public void start()
         throw new ISE("can't start.");
       }
 
-      log.info("Starting HttpServerInventoryView.");
+      log.info("Starting %s.", execNamePrefix);
```
Comment (abhishekagarwal87) [inline server/src/main/java/org/apache/druid/client/HttpServerInventoryView.java:138]: if you are changing here, there are few more places in this class that need to be changed.

**Review by abhishekagarwal87 — APPROVED** [review]
Minor comments. LGTM otherwise. Feel free to merge if you address those.

File: server/src/main/java/org/apache/druid/client/BatchServerInventoryView.java [hunk 3]
```diff
@@ -27,59 +27,295 @@
...
+  }
+
   @Override
+  public boolean isStarted()
+  {
+    return started.get();
+  }
+
+  @Override
+  public DruidServer getInventoryValue(String containerKey)
```
[the remark this answers is not in the document]
Comment (jihoonson) [reply server/src/main/java/org/apache/druid/client/BatchServerInventoryView.java:209]: i think you are right, but am not so sure why it's named that way.

**Pull request merged**